### ISSUE TITLE
trtexec error, Error Code 3: Internal Error (SymbolAddress for getPluginCreators could not be loaded,

### ISSUE THREAD

**Issue by jjxyai** [NVIDIA/TensorRT#4051]
## Description
convert onnx to engine with trtexec:

`trtexec --onnx=model.onnx --dynamicPlugins=plugin_lib.dll  --saveEngine=model16.engine --fp16 --allowGPUFallback`

with error output:

> [E] [TRT] 3: [libLoader.cpp::nvinfer1::rt::DynamicLibrary::Impl::untypedSymbolAddress::365] Error Code 3: Internal Error (SymbolAddress for getPluginCreators could not be loaded, check function name against library symbol)

## Environment

**TensorRT Version**:8.6

**NVIDIA GPU**:4060

**NVIDIA Driver Version**:546.92 

**CUDA Version**:12.3

**CUDNN Version**:8.9

Operating System:win11

**Comment by lix19937** [comment]
Check your dll function api right, or use static plugin.     
`dumpbin /exports  ./plugin_lib.dll`

**Comment by leimao** [comment]
[`getCreators`](https://github.com/NVIDIA/TensorRT/blob/4575799a91f67c060cd34212a9a27d0264460071/plugin/common/vfcCommon.h#L38) (previously known as `getPluginCreators`) have to be implemented in the plugin shared library.

**Comment by ttyio** [comment]
yes, this is documented in https://docs.nvidia.com/deeplearning/tensorrt/developer-guide/index.html#generating-plugin-library, will close this, thanks all!

**Comment by ttyio** [comment]
> > https://docs.nvidia.com/deeplearning/tensorrt/developer-guide/index.html#generating-plugin-library
> 
> No, it did not say it is a must for trtexec.

This is a must if you want your plugin work with the version compatibility feature. However, I reopen this created internal issue to track the document update for this.

**Comment by jjxyai** [comment]
> [`getCreators`](https://github.com/NVIDIA/TensorRT/blob/4575799a91f67c060cd34212a9a27d0264460071/plugin/common/vfcCommon.h#L38) (previously known as `getPluginCreators`) has to be implemented in the plugin shared library if you want to use trtexec `--dynamicPlugins`.

solved by adding methods mentioned in [9.9.1. Generating Plugin Shared Libraries](https://docs.nvidia.com/deeplearning/tensorrt/developer-guide/index.html#generating-plugin-library)

thanks a lot!!
